### PR TITLE
add support for using fallback 'arch=*' key in dependency version specified as arch->version mapping

### PR DESCRIPTION
**Pull request by boegel**
follow-up for #3506

### CODE REVIEW

File: test/framework/easyconfig.py [hunk 1]
```diff
@@ -305,6 +307,68 @@ def test_dependency(self):
...
+            'description = "test easyconfig"',
+            'toolchain = {"name":"GCC", "version": "4.6.3"}',
+            'dependencies = ['
+            '   ("not_on_power", {"arch=*": "1.2.3", "arch=POWER": False}),',
+            ']',
+        ])
+        self.prep()
+
+        # only non-POWER arch, dependency is retained
+        for arch in (AARCH64, X86_64):
```
[the remark this answers is not in the document]
Comment (boegel) [reply test/framework/easyconfig.py:360]: Oh nice catch Hound!

File: test/framework/easyconfig.py [hunk 2]
```diff
@@ -305,6 +307,69 @@ def test_dependency(self):
...
+            'description = "test easyconfig"',
+            'toolchain = {"name":"GCC", "version": "4.6.3"}',
+            'dependencies = ['
+            '   ("first", "1.0"),',
+            '   ("second", {"arch=%s": False}),' % arch,
+            ']',
+            'builddependencies = [',
+            '   ("first_build", {"arch=%s": False}),' % arch,
+            '   ("second_build", "2.0"),',
+            ']',
```
Comment (akesandgren) [inline test/framework/easyconfig.py:335]: Style nitpicking, we usually keep builddeps before deps, and if would make it less confusing regarding the assert checks below where second_build comes before first in the deps array.

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Review by akesandgren — APPROVED** [review]
LGTM

File: test/framework/easyconfig.py (hunk 2 above)
Comment (boegel) [reply test/framework/easyconfig.py:331]: fixed in 3c1eb7f

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**